app.go

```diff
@@ -680,8 +692,10 @@ func (a *Application) processTx(txBytes []byte, isCheckTx bool) (TxHandlerResult
 
 	r, err := a.TxHandler.ProcessTx(state, txBytes, isCheckTx)
 	if err != nil {
+		if !isCheckTx {
+			receiptHandler.CommitCurrentReceipt()
```

Comment:
What did we just learn about changing the number of receipts that end up in the receipts cache?

Reply:
Oops, not much apparently, fixed